### DISCUSSION TITLE
Consider using "hw_version" as part of the discovery

### DISCUSSION THREAD

**Discussion by badnetmask** [andrew-codechimp/HA-Battery-Notes#1271] [Ideas]
I have a few Somfy shades which, for whatever reason, shows the battery as a separate device from the shade in Home Assistant ([Overkiz integration](https://www.home-assistant.io/integrations/overkiz/)), but the problem is that both devices have the same model and manufacturer. Battery Notes deal with this device by identifying it as ["manual" type in the library](https://github.com/andrew-codechimp/HA-Battery-Notes/blob/main/custom_components/battery_notes/data/library.json#L3231).

I was looking at `core.device_registry` and realized that the difference between the two devices is the `hw_version`. The shade itself shows up as `"hw_version": "zigbee:SomfyRollerShadeComponent"`, while the battery shows up as `"hw_version": "zigbee:SomfyConfigurationToolComponent"`.

Would it be possible to add `hw_version` as an extra data point to take into consideration when detecting those corner cases?

To be honest, I don't know how often that happens, or how easy it would be to implement, and that's the reason why I am posting here as an idea, rather than an issue. Also, right now I'm lacking a little bit of mental space to set up a dev environment, so I am unable to directly propose a piece of code to set this up (which I usually would under normal circumstances). So, if you could at least let me know if it's worth the trouble, I could work to figure this out (eventually).

Thank you.

**Comment by andrew-codechimp** [comment]
That's interesting, I've never seen hw_version used like that but wonder if some of the other manual types also have this hidden differentiator, just checked my own device_registry and I'm seeing some occasional version numbers but mostly nulls. 
Is it displayed anywhere in the UI device screen?

It would be quite a lot of work, changing the schema adding it as optional, making sure it didn't break old versions, then adding it as an optional extra comparison when searching the library which is actually the easy bit.

Then decide if it would confuse too many people if added to the device issue form or just rely on manual PR's for these specific devices.

Worth the trouble? I doubt many people would know how to find this if it's not in the UI to actually add devices so probably not but I've done edge cases before so I'll not dismiss it but keep it in the back of my mind.

And thank you for the coffees ❤️

**Comment by badnetmask** [comment]
Yes, the hardware version is visible in the UI.
![image](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/1847440/8bdc5200-bbff-40fe-8246-1033615fd99d)

Thanks for confirming it's not an absurd idea.

**Reply by andrew-codechimp** [reply]
That certainly increases the potential of users adding these so it goes up on my list.
I'll have to try out the schema change to see how I can at least add the extra field without breaking everyone.  I've not done a library schema update before and always get scared when I know I'm potentially effecting 5k+ users but knew it would come one day 😄

**Reply by andrew-codechimp** [reply]
I've spent a little time already on this and the schema change doesn't break things so getting more comfortable with it, just need to change the library lookup code, depending on life/work it might go into beta this week.

Can you supply the details you'd want in these two devices as below so I can add them when I merge this.

```
        {
            "manufacturer": "Espressif",
            "model": "m5stack-atom",
            "hw_version": "test",
            "battery_type": "Rechargeable",
            "battery_quantity": 2
        },
```

**Reply by badnetmask** [reply]
```
{
  "manufacturer": "SOMFY",
  "model": "Sonesse 28 WF Li-Ion Roller",
  "hw_version": "zigbee:SomfyConfigurationToolComponent",
  "battery_type": "Rechargeable",
  "battery_quantity": 1
},
```
Thank you!

**Reply by andrew-codechimp** [reply]
And the zigbee:SomfyRollerShadeComponent as well please.

**Reply by badnetmask** [reply]
It's exactly the same thing, and that's the whole problem. :smile: 
```
{
  "manufacturer": "SOMFY",
  "model": "Sonesse 28 WF Li-Ion Roller",
  "hw_version": "zigbee:SomfyRollerShadeComponent",
},
```

**Reply by andrew-codechimp** [reply]
But what is the battery_type & quantity for this SomfyRollerShadeComponent?

**Comment by badnetmask** [comment]
It doesn't have a battery. That's the shade itself. The other one is the
battery. Even though they are physically one equipment, they show up in the
integration as two entities.


Mauricio Teixeira
(sent from mobile, sorry for my brevity)

**Reply by andrew-codechimp** [reply]
Ah ok, so the SomfyRollerShadeComponent should never be discovered/suggested. I was expecting two different battery types in both bits but what I'm doing will still work.
Thanks for the confirmation.

**Comment by andrew-codechimp** [comment]
I've just released this in 2.2.2 beta along with the appropriate library entries for your two devices.
If you haven't already you'll have to enable betas within HACS for battery notes.

**Reply by badnetmask** [reply]
I didn't even have to load the beta code. As soon as I entered my HA UI it said new devices were discovered. However, the discovered devices are the shades themselves, not the batteries.

**Reply by andrew-codechimp** [reply]
I think that's because the new schema for hw_version devices will do weird things with the existing code.
The new code should work out the batteries for them correctly.
2.2.3 just released, major bug I missed stopped it loading on my prod system!

**Reply by badnetmask** [reply]
Yes! After installing 2.2.3 and restarting, it correctly detected the battery! Great job! Thank you!

**Reply by andrew-codechimp** [reply]
Thanks for the confirmation, I was faking it on my dev system so wasn't 100% sure.
I'll see if I get reports of any unexpected issues and release to prod later this week or early next.
